### PR TITLE
Move project metadata and remove setup.py

### PR DESCRIPTION
**Pull request by alexfikl**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,77 @@
...
+    "jax>=0.4",
+]
+pyopencl = [
+    "islpy>=2024.1",
+    "loopy>=2024.1",
+    "pyopencl>=2024.1",
+]
+pytato = [
+    "pytato>=2021.1",
+]
```
Comment (alexfikl) [inline pyproject.toml:49]: I added all of the various array context implementations as optional dependencies here. Let me know if you prefer them somewhere else!

Comment (alexfikl) [reply pyproject.toml:49]: Potentially "fixes" #247 if left like this :grin:

Comment (inducer) [reply pyproject.toml:49]: Putting those there seems OK. As for #247, there's still `call_loopy`, which is a bit questionable.

Comment (alexfikl) [reply pyproject.toml:49]: Yeah, the code itself is still quite dependent on loopy, at least morally :(

**Pull request merged**